azure/worker/type_converters.py

```diff
@@ -107,3 +107,46 @@ def from_proto(cls, data: protos.TypedData) -> typing.Any:
         info = json.loads(data.json)
         return type_impl.TimerRequest(
             past_due=info.get('IsPastDue', False))
+
+
+class BlobConverter(type_meta.InConverter,
+                    type_meta.OutConverter,
+                    binding=type_meta.Binding.blob):
+
+    @classmethod
+    def check_python_type(cls, pytype: type) -> bool:
+        return (issubclass(pytype, (str, bytes, bytearray,
+                                    azf.InputStream) or
+                hasattr(pytype, 'read')))
```

Comment:
Also check if `pytype.read` is a callable? Why is duck typing a good idea here?

Reply:
Duck typing is to support `file-like`.  There's no standard ABC for that, but I can add one if you think that would be cleaner.